code.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    while i < self.length { |i| yield(self[i]) }
+  end
+
+  def my_each_with_index
+    while i < self.length { |i| yield(self[i], i) }
+  end
+
+  def my_select
+    result = []
+    self.my_each { |i| result.push(i) if yield(i) }
+    result
+  end
```

Comment:
This too as in the comment for `my_each` on returning `Enumerator` when no block is given.

See ruby doc on [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select) behavior.